clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/QueryTool.java

```diff
@@ -80,55 +148,301 @@ public static Map<String, String> queryStoreForKey(
+          (AvroGenericReadComputeStoreClient<Object, Object>) client;
+      AvroComputeAggregationRequestBuilder<Object> builder =
+          new AvroComputeAggregationRequestBuilder<>(computeStoreClient, ClientFactory.getSchemaReader(clientConfig));
+      builder.countGroupByBucket(bucketPredicates, fields);
+
+      ComputeAggregationResponse response = builder.execute(keys).get(60, TimeUnit.SECONDS);
+
+      outputMap.put("query-type", "countByBucket");
+      outputMap.put("keys", keyString);
+      outputMap.put("fields", countByBucketFields);
+      outputMap.put("bucket-definitions", bucketDefinitions);
+
+      // Add results for each field
+      for (String field: fields) {
+        Map<String, Integer> bucketCounts = response.getBucketNameToCount(field);
+        outputMap.put(field + "-bucket-counts", bucketCounts.toString());
+      }
+
+      return outputMap;
+    }
+  }
+
+  public static Set<Object> parseKeys(String keyString, AvroGenericStoreClient<Object, Object> client)
+      throws Exception {
+    Set<Object> keys = new HashSet<>();
+    String[] keyStrings = keyString.split(",");
+
+    AbstractAvroStoreClient<Object, Object> castClient =
+        (AbstractAvroStoreClient<Object, Object>) ((StatTrackingStoreClient<Object, Object>) client)
+            .getInnerStoreClient();
+    Schema keySchema = castClient.getKeySchema();
+
+    // Transfer vson schema to avro schema.
+    while (keySchema.getType().equals(Schema.Type.UNION)) {
+      keySchema = VsonAvroSchemaAdapter.stripFromUnion(keySchema);
+    }
+
+    for (String keyStr: keyStrings) {
+      Object key = convertKey(keyStr.trim(), keySchema);
+      keys.add(key);
+    }
+
+    return keys;
+  }
+
+  public static Map<String, Predicate<Integer>> parseBucketDefinitions(String bucketDefinitions) {
+    Map<String, Predicate<Integer>> bucketPredicates = new HashMap<>();
+
+    if (bucketDefinitions == null || bucketDefinitions.isEmpty()) {
+      return bucketPredicates;
+    }
+
+    String[] bucketDefs = bucketDefinitions.split(",");
+    for (String bucketDef: bucketDefs) {
+      bucketDef = bucketDef.trim();
+
+      // Check if it's a range format (e.g., "20-25")
+      if (bucketDef.contains("-")) {
+        String[] range = bucketDef.split("-");
+        if (range.length != 2) {
+          throw new VeniceException("Invalid range format: " + bucketDef + ". Expected format: min-max");
+        }
+
         try {
-          key = new GenericDatumReader<>(keySchema, keySchema).read(
-              null,
-              AvroCompatibilityHelper.newJsonDecoder(keySchema, new ByteArrayInputStream(keyString.getBytes())));
-        } catch (IOException e) {
-          throw new VeniceException("Invalid input key:" + keyString, e);
+          int min = Integer.parseInt(range[0].trim());
+          int max = Integer.parseInt(range[1].trim());
+
+          // Create a predicate for the range [min, max]
+          Predicate<Integer> predicate =
```

Comment:
Seems this is only for `IntPredicate`, we may need to think if we need to support all of Predicates, and if we decide to do this in the future, we should add java doc or tool usage doc we only support `IntPredicate` for now.